design/definitions.go

```diff
@@ -652,9 +664,12 @@ func (r *ResourceDefinition) DSL() func() {
 }
 
 // Finalize is run post DSL execution. It merges response definitions, creates implicit action
-// parameters, initializes querystring parameters and sets path parameters as non zero attributes.
+// parameters, initializes querystring parameters, sets path parameters as non zero attributes
+// and sets the fallbacks for security methods.
 func (r *ResourceDefinition) Finalize() {
 	r.IterateActions(func(a *ActionDefinition) error {
+		a.Finalize()
```

Comment:
Oh whoops, funny how one can be more permissive with his own code :P
If it's simple I think DefinitionSet make more sense.